web/crux/src/app/notification/notification.dto.ts

```diff
@@ -0,0 +1,78 @@
+import { ApiProperty } from '@nestjs/swagger'
+import { IsBoolean, IsIn, IsString, IsUrl } from 'class-validator'
+
+export const NOTIFICATION_TYPE_VALUES = ['discord', 'slack', 'teams'] as const
+export type NotificationTypeDto = (typeof NOTIFICATION_TYPE_VALUES)[number]
+
+export const NOTIFICATION_EVENT_TYPE_VALUES = [
+  'deployment-created',
+  'version-created',
+  'node-added',
+  'user-invited',
+] as const
+export type NotificationEventTypeDto = (typeof NOTIFICATION_EVENT_TYPE_VALUES)[number]
+
+export class NotificationDto {
+  id: string
+
+  name: string
+
+  url: string
+
+  @ApiProperty({ enum: NOTIFICATION_TYPE_VALUES })
```

Comment:
Why not? 

```
  @IsString()
  @IsIn(NOTIFICATION_EVENT_TYPE_VALUES)
  ```

Reply:
I beleive `@IsString()` is not necessary

Reply:
Yep, works without `@IsString()`.